src/Proxy/DriveItemProxy.php

```diff
@@ -255,45 +257,86 @@ public function delete()
-
-        $response = $this
+        $uploadSessionEndpoint = "$driveLocator$itemLocator:/$name:/createUploadSession";
+        //include extended options with existing item for upload session
+        $item = array_merge(
+            array("name" => $name, 'description' => '' ,"@microsoft.graph.conflictBehavior" => "rename"), $options
+        );
+
+		//create an upload session for large files
+		$uploadSessionResponse = $this
             ->graph
-            ->createRequest('PUT', $endpoint)
-            ->addHeaders($options)
-            ->attachBody($body)
+            ->createRequest('POST', $uploadSessionEndpoint)
+            ->addHeaders(array('Content-Type' => 'application/json'))
+            ->attachBody(array("item" =>$item))
             ->execute();
-
-        $status = $response->getStatus();
-
-        if ($status != 201) {
-            throw new \Exception("Unexpected status code produced by 'PUT $endpoint': $status");
-        }
-
-        $driveItem = $response->getResponseAsObject(DriveItem::class);
-
-        return new self($this->graph, $driveItem);
+		
+		$uploadSessionResponseBody=$uploadSessionResponse->getBody();
+
+        if(is_array($uploadSessionResponseBody) && array_key_exists("uploadUrl", $uploadSessionResponseBody)) {
+
+            $uploadUrl = $uploadSessionResponseBody['uploadUrl'];
+
+			$handle = @fopen($name, "r");
+			if ($handle) {
+				$bytesRead=0;
+				$buffer = "";
+				$totalSize=filesize($name);
+				
+				while (!feof($handle)) {
+					$buffer = fread($handle, CHUNK_SIZE);
+					$contentLength = strlen($buffer);
+					ob_flush(); //flush output buffer high-level
+					flush(); //flush output buffer low-level
+					
+					//initialise bytes read, and interval range
+					$byteIntervalStart=$bytesRead;
+					$bytesRead += $contentLength;
+					$byteIntervalEnd=$bytesRead-1;
+					
+					//upload headers containing bytes Content length and range
+					$uploadSessionHeaders = array();
+					$uploadSessionHeaders['Content-Type'] = 'application/json';
+					$uploadSessionHeaders['Content-Length'] = $contentLength;
+					$uploadSessionHeaders['Content-Range'] = 'bytes ' . $byteIntervalStart . "-" . $byteIntervalEnd . "/" . $totalSize;
+					
+					//create a put request for chunks
+					$chunkedResponse = $this
+		                 ->graph
+		                 ->createRequest('PUT', $uploadUrl)
+		                 ->addHeaders($uploadSessionHeaders)
+		                 ->attachBody($buffer)
+		                 ->execute();
+					
+					$status = $chunkedResponse->getStatus();
+					
+					if($status != 201 && $status != 202){
+						throw new \Exception("Unexpected status code produced by 'PUT $endpoint': $status");
+					}
+					
+					if($status === 201){
+						//resource created in onedrive return response object as DriveItem
+						return $chunkedResponse->getResponseAsObject(DriveItem::class);
+					}
+                }
+			}
+			fclose($handle);
+		}
+		
+		return NULL;
```

Comment:
Why not returning something useful here? The created DriveItem would be expected.

Also, please do not uppercase null: `null` is what we use in this project.

Reply:
done